tests/CMakeLists.txt

```diff
@@ -43,5 +47,5 @@ foreach(testfile IN LISTS TESTSRC)
   find_package(OpenMP REQUIRED)
   target_link_libraries(${test_name} PRIVATE OpenMP::OpenMP_Fortran)
 
-  add_test(NAME ${test_name} COMMAND sh -c "mpirun -np 1 ${test_name}")
+  add_test(NAME ${test_name} COMMAND sh -c "mpirun --oversubscribe -np ${CMAKE_CTEST_NPROCS} ${test_name}")
```

Comment:
Allocator and reorder tests are the ones that doesn't require a parallel run. Both of them don't do any communication and even in a full simulation with multiple ranks they're always local to their own rank. Running these on multiple ranks is still a valid test but if we can avoid this easily it is better I guess.

Reply:
that's a good point, though I can't think of a clean way of doing it yet.

Reply:
You could use a function to build the test definition, I'll see if I can dig out an example where I've done this

Reply:
Something like the following should work
```
function(define_test test_name np)
  add_test(NAME ${test_name} COMMAND sh -c "mpirun --oversubscribe -np ${np} ...")
endfunction()

# Make a single process test (assumes there is a test_foo target)
define_test(test_foo 1)

# Make a parallel test
define_test(test_bar ${CMAKE_CTEST_NPROCS})
```